### PR TITLE
Add comprehensive test suite for ock CLI

### PR DESCRIPTION
**Pull request by williamwmarx**
## Summary
- Add unit tests for cli, selector, main, and utils modules
- Add integration tests covering real-world usage scenarios  
- Document existing bug with step values being incorrectly decremented

## Test Coverage
The test suite covers:
- Input parsing (files, stdin, literal text)
- Selector parsing (indices, ranges, steps, regex patterns)
- Row and column selection logic
- Various delimiters (whitespace, comma, tab, custom)
- Edge cases (empty input, out-of-bounds, Unicode, special characters)
- Real-world scenarios (ps aux simulation, CSV processing)

## Known Issues
Tests have identified a bug in selector.rs:96 where step values are incorrectly decremented by 1. This will be addressed in a separate PR.

## Test Plan
- [x] All existing functionality remains unchanged
- [x] Unit tests pass (except for known step bug)
- [x] Integration tests pass (except for step-related test)
- [x] No changes to production code logic (only test visibility)

### CODE REVIEW

File: src/selector_tests.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        assert_eq!(selectors[0].step, 1);
+    }
+
+    #[test]
+    fn test_parse_selectors_range_with_step() {
+        let selectors = parse_selectors(&String::from("1:10:2"));
+        assert_eq!(selectors.len(), 1);
+        assert_eq!(selectors[0].start_idx, 0); // 1-1 (correct: convert to 0-based index)
+        assert_eq!(selectors[0].end_idx, 9);   // 10-1 (correct: convert to 0-based index)
+        assert_eq!(selectors[0].step, 2);      // BUG: Expected 2, but gets 1 due to incorrect subtraction
```
[the remark this answers is not in the document]
Comment (williamwmarx) [reply src/selector_tests.rs:56]: known issue. being fixed now.
